src/main/java/org/jenkins_ci/plugins/run_condition/core/StringsMatchCondition.java

```diff
@@ -32,17 +32,26 @@
 import org.jenkinsci.plugins.tokenmacro.TokenMacro;
 import org.kohsuke.stapler.DataBoundConstructor;
 
+import java.util.Map;
+import java.util.logging.Logger;
+
 public class StringsMatchCondition extends AlwaysPrebuildRunCondition {
 
     final String arg1;
     final String arg2;
     final boolean ignoreCase;
+    final boolean shellVariables;
 
     @DataBoundConstructor
-    public StringsMatchCondition(final String arg1, final String arg2, final boolean ignoreCase) {
+    public StringsMatchCondition(final String arg1, final String arg2, final boolean ignoreCase,final boolean shellVariables) {
         this.arg1 = arg1;
         this.arg2 = arg2;
         this.ignoreCase = ignoreCase;
+        this.shellVariables=shellVariables;
+    }
+
```

Comment:
Use a `DataBoundSetter` rather than modifying the `DataBoundConstructor` so that the signature of the constructor does not change.
```suggestion

    @org.kohsuke.stapler.DataBoundSetter
    public void setShellVariables(boolean value) {
        shellVariables = value;
    }
```

Reply:
@MarkEWaite Okay Sir

Reply:
@MarkEWaite Done Sir